### ISSUE TITLE
Sample config file isn't in a working format

### ISSUE THREAD

**Issue by rbubley** [dask/distributed#2419]
With distributed 1.25.1, if you start up with no configuration file, one is created for you in: `~/.config/dask/distributed.yaml`

This starts:
```
# distributed:
#   version: 2
#   # logging:
#   #   distributed: info
#   #   distributed.client: warning
#   #   bokeh: critical
```

If you were to remove the hashes and try to use the config file as is, it doesn't work -- the code still expects the old-style config files., for example, dask.security does not look under the key `distributed.comm`. for `tls`, it expects it at the top level.

**Comment by mrocklin** [comment]
Thank you for raising this @rbubley .  What do you think is best here?  Should we change the code in `distributed/security.py` to look in the proper place?  Do you have any interest in submitting a patch here?

**Comment by rbubley** [comment]
I'm afraid I'm not likely to have time to work on this any time soon, but thought it might be useful to note in case other people were to spend time debugging too. Can I suggest as a super-easy workaround until someone has time to address this properly is to simply document it by adding a comment to `distributed/distributed/distributed.yaml`:

```
# Dask Distributed currently uses the original configuration format for the security options; please see https://distributed.dask.org/en/latest/configuration.html
```

**Comment by mrocklin** [comment]
This seems easy enough to fix in code that we should probably wait for
that.  Marking as a good first issue.

**Comment by mrocklin** [comment]
In particular if you search distributed/security.py for `dask.config.config` you'll find that this dictionary is passed into some function that hopefully grabs the relevant information from it.  My guess is that we instead need to pass in `dask.config.get('distributed.comm.tls')` or something similar (maybe `dask.config.get('distributed.comm')` or something else).  

I don't personally have a good way to test this, but @rbubley might have suggestions.

**Comment by rbubley** [comment]
The way I noticed this is that although I had set tls options, and `require-encryption`  in the config file, `dask-scheduler` still published a tcp adddress when it started up. I expected it to publish a tls address. An equivalent could be done programatically in a test.

**Comment by VMois** [comment]
@rbubley Have you changed `default-scheme` option to use `tls`?

**Comment by rbubley** [comment]
Hi @realeyes, the issue is not what the options selected are, it’s that security.py isn’t looking in the right place to read them. (By right place, I mean in the format that the auto-created distributed.yaml tells you to use.) 

Basically, when distributed moved over to using the new dask config approach, security.py wasn’t updated for it.

**Comment by prasunanand** [comment]
Hi,

Is anyone working on it ? :)

**Comment by jrbourbeau** [comment]
I don't think so. However, this issue may have already been fixed by https://github.com/dask/distributed/pull/2873. cc @jcrist 

@prasunanand as a side note, all conversation around Dask development takes place on GitHub so if there's not a comment indicating that someone is working on an issue, then feel free to jump in

**Comment by jcrist** [comment]
#2873 fixed this for security, https://github.com/dask/distributed/pull/2952 (still open, needs review) fixes this for logging. I think this issue can be closed, #2937 is more specific for the current state.
